app/src/main/java/io/homeassistant/companion/android/bluetooth/ble/TransmitterManager.kt

```diff
@@ -0,0 +1,81 @@
+package io.homeassistant.companion.android.bluetooth.ble
+
+import android.bluetooth.BluetoothAdapter
+import android.bluetooth.le.AdvertiseCallback
+import android.bluetooth.le.AdvertiseSettings
+import android.content.Context
+import org.altbeacon.beacon.Beacon
+import org.altbeacon.beacon.BeaconParser
+import org.altbeacon.beacon.BeaconTransmitter
+object TransmitterManager {
+    private lateinit var physicalTransmitter: BeaconTransmitter
+    private lateinit var beacon: Beacon
+
+    private fun buildBeacon(haTransmitter: BLETransmitter): Beacon {
+        val builder = Beacon.Builder()
+        builder.setTxPower(-59)
+        builder.setId1(haTransmitter.id1)
+        builder.setId2(haTransmitter.id2)
+        builder.setId3(haTransmitter.id3)
+        builder.setManufacturer(0x004c) //  apple ibeacon
```

Comment:
Does this mean we are hard coded only to ibeacon?

Reply:
Yes, for now. We can expand to other formats if required later, but i think for most use cases one beacon type is fine, and iBeacon appears to be the most popular.